conformance/conformance.go

```diff
@@ -0,0 +1,230 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package conformance contains the core setup and execution logic
+// for the Gateway API Inference Extension conformance test suite.
+package conformance
+
+import (
+	"fmt"
+	"io/fs"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	clientset "k8s.io/client-go/kubernetes"
+
+	// Import runtime package for scheme creation
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+	"sigs.k8s.io/yaml"
+
+	// Import necessary types and utilities from the core Gateway API conformance suite.
+	// Assumes sigs.k8s.io/gateway-api is a dependency in the go.mod.
+	gatewayv1 "sigs.k8s.io/gateway-api/apis/v1"            // Import core Gateway API types
+	confapis "sigs.k8s.io/gateway-api/conformance/apis/v1" // Report struct definition
+	confconfig "sigs.k8s.io/gateway-api/conformance/utils/config"
+	confflags "sigs.k8s.io/gateway-api/conformance/utils/flags"
+	confsuite "sigs.k8s.io/gateway-api/conformance/utils/suite"
+	"sigs.k8s.io/gateway-api/pkg/features" // Using core features definitions if applicable
+
+	// Import the test definitions package to access the ConformanceTests slice
+	"sigs.k8s.io/gateway-api-inference-extension/conformance/tests"
+
+	// Import test packages using blank identifier
+	// This triggers the init() functions in these packages, which register the tests
+	// by appending them to the tests.ConformanceTests slice.
+	_ "sigs.k8s.io/gateway-api-inference-extension/conformance/tests/basic"
+	// TODO: Add blank imports for other test categories as they are created.
+	// _ "sigs.k8s.io/gateway-api-inference-extension/conformance/tests/model_routing"
+
+	// Import the Inference Extension API types
+	inferencev1alpha2 "sigs.k8s.io/gateway-api-inference-extension/api/v1alpha2"
+)
+
+// GatewayLayerProfileName defines the name for the conformance profile that tests
+// the Gateway API layer aspects of the Inference Extension (e.g., InferencePool, InferenceModel CRDs).
+// Future profiles will cover EPP and ModelServer layers.
+const GatewayLayerProfileName confsuite.ConformanceProfileName = "Gateway"
+
+var InferenceCoreFeatures = sets.New[features.FeatureName]() // Placeholder - Populate with actual features specific to this profile or manage features per profile
+
+// GatewayLayerProfile defines the conformance profile for the Gateway API layer
+// of the Inference Extension.
+// In future iterations, we will add constants and ConformanceProfile structs for
+// EPPProfileName ("EPP") and ModelServerProfileName ("ModelServer")
+// to cover their respective conformance layers.
+var GatewayLayerProfile = confsuite.ConformanceProfile{
```

Comment:
The term "layer" leads me to believe conformance dependencies exist between EPP, Gateway, etc. @robscott does Gateway API have a naming standard for different profiles? IMO, using "component" indicates independence among the profiles.